src/MongoDB.EntityFrameworkCore/Metadata/BsonRepresentationConfiguration.cs

```diff
@@ -0,0 +1,85 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using MongoDB.Bson;
+
+namespace MongoDB.EntityFrameworkCore.Metadata;
+
+/// <summary>
+/// Configuration of a BSON representation defining how the value from property
+/// should be stored in the MongoDB database.
+/// </summary>
+public record BsonRepresentationConfiguration
```

Comment:
public readonly record?

Reply:
public readonly record struct?

Reply:
Can only use readonly if we change it to struct. Given that this is an optional/nullable in a bunch of places that means you won't be able to access .BsonType without going through .Value everywhere.

e.g. in the tests

```csharp
Assert.NotNull(representation);
Assert.Equal(bsonType, representation.BsonType);
Assert.Equal(allowOverflow, representation.AllowOverflow);
Assert.Equal(allowTruncation, representation.AllowTruncation);
```

Becomes

```csharp
Assert.NotNull(representation);
Assert.Equal(bsonType, representation.Value.BsonType);
Assert.Equal(allowOverflow, representation.Value.AllowOverflow);
Assert.Equal(allowTruncation, representation.Value.AllowTruncation);
```

Which I don't think is an improvement.